src/Refinery/KindlyTo/Transformation/DateTimeTransformation.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/* Copyright (c) 2020 Luka K. A. Stocker, Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\Refinery\KindlyTo\Transformation;
+
+use ILIAS\Refinery\ConstraintViolationException;
+use ILIAS\Refinery\DeriveApplyToFromTransform;
+use ILIAS\Refinery\Transformation;
+
+/**
+ * Transform date format to DateTimeImmutable
+ * Please note:
+ * - RFC3339 & W3C format output on screen is the same as Atom
+ * - RFC850 format output on screen is the same as Cookie
+ * - RFC1036, RFC1123, RFC2822 & RSS format output on screen is the same as RFC822
+ */
+class DateTimeTransformation implements Transformation
+{
+    use DeriveApplyToFromTransform;
+
+    /**
+     * @inheritdoc
+     */
+    public function transform($from)
+    {
+        $formats = [
+            \DateTimeImmutable::ATOM,
+            \DateTimeImmutable::COOKIE,
+            \DateTimeImmutable::ISO8601,
+            \DateTimeImmutable::RFC822,
+            \DateTimeImmutable::RFC7231,
+            \DateTimeImmutable::RFC3339_EXTENDED
+        ];
+
+        if (is_string($from)) {
+            foreach ($formats as $format) {
+                $res = \DateTimeImmutable::createFromFormat($format, $from);
+                if ($res instanceof \DateTimeImmutable) {
+                    return $res;
+                }
+            }
+        }
+
+        if (is_int($from) || is_float($from)) {
+            $UnixTimestamp = strtotime($from);
```

Comment:
According to https://www.php.net/manual/en/function.strtotime.php `strtotime` accepts only argumens of type `string`. Here, you pass either an `int` or a `float`.

This will result in a `PHP Fatal error:  Uncaught TypeError: strtotime() expects parameter 1 to be string, int given` if strict types are enabled.

The interface description defines: `integers, which will be interpreted as Unix timestamps.`

When reading this, I would expect the following code:

```php
if (is_int($from)) {
    return new \DateTimeImmutable('@' . $from);
}
```

Reply:
Nice catch. Thanks!

Reply:
:eyes: FYI: I did not change this, yet. So if you agree, we'll have to change this (you can pass this to me if you like).

Reply:
Yeah, I know =) I'm currently on it.